modules/kafka/src/main/java/org/testcontainers/containers/KafkaContainer.java

```diff
@@ -14,38 +14,51 @@
  *
  */
 public class KafkaContainer extends GenericContainer<KafkaContainer> {
+    public static final int DEFAULT_KAFKA_PORT = 9093;
 
-    public static final int KAFKA_PORT = 9093;
+    public static final String DEFAULT_KAFKA_BROKER_ID = "1";
+
+    public static final int DEFAULT_INTERNAL_TOPIC_RF = 1;
 
     public static final int ZOOKEEPER_PORT = 2181;
 
+    public static final String CONFLUENT_PLATFORM_VERSION = "5.2.1";
+
     private static final int PORT_NOT_ASSIGNED = -1;
 
     protected String externalZookeeperConnect = null;
 
     private int port = PORT_NOT_ASSIGNED;
+    private int exposedPort;
 
     public KafkaContainer() {
-        this("5.2.1");
+        this(CONFLUENT_PLATFORM_VERSION);
     }
 
     public KafkaContainer(String confluentPlatformVersion) {
+        this(confluentPlatformVersion, DEFAULT_KAFKA_PORT, DEFAULT_KAFKA_BROKER_ID, DEFAULT_INTERNAL_TOPIC_RF);
+    }
+
+    public KafkaContainer(String confluentPlatformVersion, int exposedPort, String brokerId, int internalTopicRf) {
         super(TestcontainersConfiguration.getInstance().getKafkaImage() + ":" + confluentPlatformVersion);
 
+        this.exposedPort = exposedPort;
         // TODO Only for backward compatibility
         withNetwork(Network.newNetwork());
         withNetworkAliases("kafka-" + Base58.randomString(6));
-        withExposedPorts(KAFKA_PORT);
+        withExposedPorts(exposedPort);
 
         // Use two listeners with different names, it will force Kafka to communicate with itself via internal
         // listener when KAFKA_INTER_BROKER_LISTENER_NAME is set, otherwise Kafka will try to use the advertised listener
-        withEnv("KAFKA_LISTENERS", "PLAINTEXT://0.0.0.0:" + KAFKA_PORT + ",BROKER://0.0.0.0:9092");
+        withEnv("KAFKA_LISTENERS", "PLAINTEXT://0.0.0.0:" + exposedPort + ",BROKER://0.0.0.0:9092");
         withEnv("KAFKA_LISTENER_SECURITY_PROTOCOL_MAP", "BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT");
         withEnv("KAFKA_INTER_BROKER_LISTENER_NAME", "BROKER");
 
-        withEnv("KAFKA_BROKER_ID", "1");
-        withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "1");
-        withEnv("KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS", "1");
+        withEnv("KAFKA_BROKER_ID", brokerId);
```

Comment:
since these environment variables are set in constructor, you can override them later like this:
```java
new KafkaContainer()
    .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "3")
```

which makes me wonder whether we need to make any change to `KafkaContainer` at all.

The transaction state log, however, is something we wanted to default to 1, as seen in #1816, and can be done in this PR